pkg/tool/commands/audit.go

```diff
@@ -0,0 +1,60 @@
+package commands
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/go-kit/log"
+	"github.com/grafana/loki/v3/pkg/tool/audit"
+	util_cfg "github.com/grafana/loki/v3/pkg/util/cfg"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+// AuditIndexCommand validates an index by checking existing chunks.
+type AuditCommand struct {
+	path string
+
+	configFile string
+}
+
+func (a *AuditCommand) auditIndex(kpCtx *kingpin.ParseContext) error {
+	logger := log.NewLogfmtLogger(os.Stdout)
+
+	var auditCfg audit.Config
+	configFileArg := fmt.Sprintf("-config.file=%s", a.configFile)
+	if err := util_cfg.DefaultUnmarshal(&auditCfg, []string{configFileArg}, flag.CommandLine); err != nil {
+		fmt.Fprintf(os.Stderr, "failed parsing config: %v\n", err)
+		os.Exit(1)
+	}
+	if err := auditCfg.Validate(); err != nil {
+		fmt.Fprintf(os.Stderr, "failed validating config: %v\n", err)
+		os.Exit(1)
+	}
+
+	found, missing, err := audit.Run(context.Background(), a.path, "19859", auditCfg, logger)
```

Comment:
19859 is hardcoded. Shouldn't this be a table name?

Reply:
good question. I'll cehck ti again but my thought here was: this table name is only used by tsdb to compute the interval, which is 24h in our case. So instead of asking the user to give the table name I preset it.

That said: I'll check how the TSDB internals use it, maybe we don't even need the table at all :shrug:

Reply:
I've updated the code, WDYT?